redbot/core/utils/_internal_utils.py

```diff
@@ -316,3 +369,39 @@ async def fetch_latest_red_version_info() -> Tuple[Optional[VersionInfo], Option
         required_python = data["info"]["requires_python"]
 
         return release, required_python
+
+
+def get_cog_by_call(bot: Red, func: str) -> str:
+    try:
+        stack = inspect.stack()
+        cog_name = None
+        valid = False
+        frm = None
+        for index, item in enumerate(stack):
+            if item.function == func:
+                break
+        else:
+            index = -2
+        if index >= 0:
+            frm = stack[index + 1]
+            full_caller_page = inspect.getmodule(frm[0]).__package__
+            if full_caller_page.startswith("asyncio"):
+                return "A task, unable to pinpoint origin."
+            elif full_caller_page.startswith("redbot.cogs."):
+                cog_name = full_caller_page.split(".")[2]
+                return f"Red Core - `{cog_name}`"
+            elif full_caller_page.startswith("redbot"):
+                return "Red Core"
+            caller_package = full_caller_page.split(".")[0]
+            for cog_name, cog in bot._BotBase__cogs.items():
+                cog_package = inspect.getmodule(cog).__package__.split(".")[0]
```

Comment:
nitpick, but using the `__module__` directly looks simpler and achieves same thing.
```suggestion
                cog_package = cog.__module__.split(".")[0]
```